AK/NoAllocationGuard.h

```diff
@@ -41,7 +41,7 @@ class NoAllocationGuard {
         return Processor::current_thread()->get_allocation_enabled();
 #elif defined(AK_OS_SERENITY)
         // This extern thread-local lives in our LibC, which doesn't exist on other systems.
-        return s_allocation_enabled;
+        return __allocation_enabled;
```

Comment:
Not sure how I feel about explicitly moving towards `__`, I think we mostly do `s_` or `g_` for variables.

Reply:
sure, done :+1:

Reply:
https://clang.llvm.org/extra/clang-tidy/checks/bugprone/reserved-identifier.html

Reply:
> I suppose we could use `__g_foo` if we wanted.

I'll switch it back to `__`, without the `g`. `__` sufficiently communicates "this leaks to the outside world", otherwise it would not need a prefix to begin with.

Reply:
Done :)